src/sage/rings/semirings/tropical_mpolynomial.py

```diff
@@ -0,0 +1,390 @@
+                eq += variables[i]*e
+            eq += self.dict()[key].lift()
+            linear_eq[key] = eq
+
+        # checking for all possible combinations of two terms
+        for keys in combinations(self.dict(), 2):
+            sol = solve(linear_eq[keys[0]]==linear_eq[keys[1]], variables)
+            
+            # parametric solution of the chosen two terms
+            final_sol = []
+            for s in sol[0]:
+                final_sol.append(s.right())
+            xy_interval = []
+            xy_interval.append(tuple(final_sol))
+            
+            # comparing with other terms
+            min_max = linear_eq[keys[0]]
+            for i,v in enumerate(variables):
+                min_max = min_max.subs(v==final_sol[i])
+            all_sol_compare = []
+            no_solution = False
+            for compare in self.dict():
+                if compare not in keys:
+                    temp_compare = linear_eq[compare]
+                    for i, v in enumerate(variables):
+                        temp_compare = temp_compare.subs(v==final_sol[i])
+                    if self.parent().base()._use_min:
+                        sol_compare = solve(min_max < temp_compare, variables)
+                    else:
+                        sol_compare = solve(min_max > temp_compare, variables)
+                    if sol_compare: # if there is solution
+                        if isinstance(sol_compare[0], list):
+                            if sol_compare[0]:
+                                all_sol_compare.append(sol_compare[0][0])
+                        else: # solution is unbounded on one side
+                            all_sol_compare.append(sol_compare[0])
+                    else:
+                        no_solution = True
+                        break
+
+            # solve the condition for parameter
+            if not no_solution:
+                parameter = set()
+                for sol in all_sol_compare:
+                    parameter = parameter.union(set(sol.variables()))
+                parameter_solution = solve(all_sol_compare, list(parameter))
+                if parameter_solution:
+                    xy_interval.append(parameter_solution[0])
+                    # calculate order
+                    index_diff = []
+                    for i in range(len(keys[0])):
+                        index_diff.append(abs(keys[0][i]-keys[1][i]))
+                    order = gcd(index_diff)
+                    xy_interval.append(order)
+                    tropical_roots.append(xy_interval)
+
+        if self.parent().ngens() == 2:
+            return TropicalCurve(self, tropical_roots)
+        else:
+            return TropicalVariety(self, tropical_roots)
+
+
+class TropicalMPolynomialSemiring(UniqueRepresentation, Parent):
+    """
+    Semiring structure of tropical polynomials in multiple variables
+    """
+
+    def __init__(self, base_semiring, n, names, order='degrevlex'):
```

Comment:
If you want term orders, use the properly implemented term orders. However, I don't think there is any utility in them. Just remove it.

Reply:
If I remove the order and that means also the `term_order()` method, then the comparisan will give an error:
```
sage: T = TropicalSemiring(QQ)
sage: R.<x,y,z> = PolynomialRing(T)
sage: x < y
Traceback (most recent call last):
...
AttributeError: 'TropicalMPolynomialSemiring_with_category' object has no attribute 'term_order'
```
How to do the properly implemented term orders as you suggest?

Reply:
Simply don't pass the term order in the constructor.